jumpscale/packages/vdc_dashboard/chats/presearch.py

```diff
@@ -0,0 +1,49 @@
+from jumpscale.sals.chatflows.chatflows import StopChatFlow, chatflow_step
+from jumpscale.packages.vdc_dashboard.sals.solutions_chatflow import SolutionsChatflowDeploy
+from jumpscale.packages.vdc_dashboard.sals.vdc_dashboard_sals import get_deployments
+
+
+class PresearchDeploy(SolutionsChatflowDeploy):
+    SOLUTION_TYPE = "presearch"
+    title = "Presearch"
+    HELM_REPO_NAME = "marketplace"
+    steps = ["get_release_name", "create_subdomain", "set_config", "install_chart", "initializing", "success"]
+
+    CHART_LIMITS = {
+        "Silver": {"cpu": "2000m", "memory": "2024Mi"},
+        "Gold": {"cpu": "3000m", "memory": "3072Mi"},
+        "Platinum": {"cpu": "4000m", "memory": "4096Mi"},
+    }
+
+    def _check_uniqueness(self):
+        username = self.user_info()["username"]
+        self.md_show_update("Preparing the chatflow...")
+        if get_deployments(self.SOLUTION_TYPE, username):
+            raise StopChatFlow("You can only have one Presearch solution per VDC")
+
+    def get_release_name(self):
+        self._check_uniqueness()
+        super().get_release_name()
+
+    def _enter_registration_code(self):
+
+        form = self.new_form()
+        self.registration_code = form.string_ask("Enter Registration Code", required=True)
+        form.ask()
+        self.registration_code = self.registration_code.value
+
+    @chatflow_step(title="Node Configuration")
+    def set_config(self):
+        self._enter_registration_code()
```

Comment:
this method is just one question, no need to make it a form
we can use something like

`self.registration_code = self.string_ask("Enter Registration Code", required=True)`

Reply:
Why can't I make a separate method even its a question ? Reason, this yet to be tested by the community and we may need to have more questions. It is fine to have a method and we can add questions later. Please explain the following,

1. How does your change impact functionality or even coding for that purpose ?
2. How does your change help code quality ? Please explicitly specify reasons in the context of this PR
3. We already have everything working properly and as expected. A link to it is here [Preseach Chatflow](https://youtu.be/X2ZAslcr4yI). 

If there is something that you think is better, please feel free to improve it ! If not, please jump on a call and we can go it over together

Reply:
> How does your change help code quality ? Please explicitly specify reasons in the context of this PR

- Form is made to ask several question, in case of one question we have other tools
- More lines of code and an extra function call while it should be only one line, that's enough reason
- When you need to do more questions you can bring it back to a form

> How does your change impact functionality or even coding for that purpose ?
> We already have everything working properly and as expected. A link to it is here Preseach Chatflow.

- Same functionality but clean code, better performance

Reply:
How have you measured performance ? Please share any execution times / performance variables and indicators when you have logged when you run my chatflow versus your chatflow ?

Reply:
What is several 2, 3 4, ?  My several could be 5 , yours could be 10 ? Where is this defined ? 

Also, please mention the tools you are using to track performance for chatflow coding ? Any samples or any statistical data previously logged before to track execution times / performance of chatflows ? 

What is "several" here ? 2, 3, 4

Reply:
No need to jump on a call, It's simple.

As @waleedhammam mentioned, no need to use a form here, as it's only single question asked with mo

> How does your change impact functionality or even coding for that purpose ?
> How does your change help code quality ? Please explicitly specify reasons in the context of this PR

Mostly we won't try to re-explain every decision taken in the course of the development of any component again,
it's not an interrogation. Some of the reviewers who worked on the same code base for years sees an improvement somewhere, you better simply follow it as they already explained why.

> We already have everything working properly and as expected. A link to it is here Preseach Chatflow.

Working doesn't imply quality.

Reply:
> How have you measured performance ? Please share any execution times / performance variables and indicators logged when you run my chatflow versus your chatflow ?

> What is several 2, 3 4, ? My several could be 5 , yours could be 10 ? Where is this defined ?
> 
> Also, please mention the tools you are using to track performance for chatflow coding ? Any samples or any statistical data previously logged before to track execution times / performance of chatflows ?

Please don't ask useless questions.

Reply:
And btw, we will no longer discuss the comments, you could've just updated the code accordingly, I think it would take less time than debating.

Reply:
For what you don't understand is useless ? 

In your code of conduct, please ensure to have fairness. Also ensure that a team should not be used to serve or settle someone's personal score.